File: zig/darts/darts.zig
pub const Coordinate = struct {
    x: f32,
    y: f32,

    pub fn init(x: f32, y: f32) Coordinate {
        return Coordinate{ .x = x, .y = y };
    }
    pub fn score(self: Coordinate) isize {
        const dist = @sqrt((self.x * self.x) + (self.y * self.y));
        if (dist > 10) {
            return 0;
        }
        if (dist > 5) {
            return 1;
        }
        if (dist > 1) {
            return 5;
        }
        return 10;
    }
};

File: vbnet/darts/Darts.vb
Imports System

Public Module Darts
  Public Function Score(ByVal x As Double, ByVal y As Double) As Integer
    Dim dist = Math.Sqrt(x*x + y*y)
    If dist <= 1
      Return 10
    ElseIf dist <= 5
      Return 5
    ElseIf dist <= 10
      Return 1
    Else
      Return 0
    End If
  End Function
End Module
